### DISCUSSION TITLE
How to get pique's user_type from PySpades connection?

### DISCUSSION THREAD

**Discussion by lokka30** [piqueserver/piqueserver#620] [Q&A] [answered]
I am currently programming an [open source script](https://github.com/lokka30/TheHallwayScripts/blob/dev/scripts/THPlayerList/THPlayerList.py) for a `/list` command, separating players by their `user_type` (member, trusted, guard, etc).

Currently on a hurdle where I require access to a variable from the piqueserver player, but I only have access to the pyspades player:
`AttributeError: 'MapExtensionConnection' object has no attribute 'user_type'`

Is it possible to get the 'rank' of a PySpades player, or will I have to consider alternatives, such as keeping track of those who use the login command successfully?

(Context: Line 19 of [this](https://github.com/lokka30/TheHallwayScripts/blob/dev/scripts/THPlayerList/THPlayerList.py) script)

**Comment by NotAFile** [comment, marked as answer]
Looks like you already fixed this. The problem was that `user_types` is a set and there is no `user_type` attribute.

**Reply by lokka30** [reply]
I apologise for not clarifying that I resolved the issue. Cheers :)
